File: backend/app/schemas/analytics.py
from pydantic import BaseModel
from typing import List, Optional
from decimal import Decimal

class CategorySpendItem(BaseModel):
    category: str
    total_amount: Decimal
    transaction_count: int
    percentage: float

class CategorySpendResponse(BaseModel):
    items: List[CategorySpendItem]
    total_spend: Decimal

class MonthlySpendItem(BaseModel):
    month: str # format: YYYY-MM
    month_label: str # e.g. "Jan 2026"
    total_amount: Decimal
    transaction_count: int

class MonthlySpendResponse(BaseModel):
    items: List[MonthlySpendItem]
    total_spend: Decimal

File: frontend/src/types/analytics.ts
export interface CategorySpendItem {
  category: string;
  total_amount: number;
  transaction_count: number;
  percentage: number;
}

export interface CategorySpendResponse {
  items: CategorySpendItem[];
  total_spend: number;
}

export interface MonthlySpendItem {
  month: string; // YYYY-MM
  month_label: string; // e.g. "Jan 2026"
  total_amount: number;
  transaction_count: number;
}

export interface MonthlySpendResponse {
  items: MonthlySpendItem[];
  total_spend: number;
}
